dxr/query.py

```diff
@@ -309,17 +320,51 @@ def generic_visit(self, node, visited_children):
         """
         return visited_children or node
 
+def filters_per_plugin(enabled_plugins):
```

Comment:
It's not clear from the docstrings what this and the next 2 functions do. Trying to figure it out....